packages/helm-charts/common/templates/_helpers.tpl

```diff
@@ -1,21 +1,74 @@
+{{/* vim: set filetype=mustache: */}}
+ {{/*
+Expand the name of the chart.
+*/}}
+{{- define "common.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "common.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := "" -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "common.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "common.standard.labels" -}}
+{{- include "common.standard.short_labels" . }}
+chart: {{ template "common.chart" . }}
+heritage: {{ .Release.Service }}
+{{- end -}}
+
+{{- define "common.standard.short_labels" -}}
+app: {{ template "common.name" . }}
+release: {{ .Release.Name }}
+{{- end -}}
+
 {{- define "common.init-genesis-container" -}}
 - name: init-genesis
   image: {{ .Values.geth.image.repository }}:{{ .Values.geth.image.tag }}
-  imagePullPolicy: {{ .Values.imagePullPolicy }}
+  imagePullPolicy: {{ .Values.geth.image.imagePullPolicy }}
+  command:
+  - /bin/sh
+  - -c
   args:
-  - "init"
-  - "/var/geth/genesis.json"
+  - |
+      mkdir -p /var/geth /root/celo
+      if [ "{{ .Values.genesis.useGenesisFileBase64 | default false }}" == "true" ]; then
+        cp -L /var/geth/genesis.json /root/.celo/
+      else
+        wget -O /root/.celo/genesis.json "https://www.googleapis.com/storage/v1/b/genesis_blocks/o/{{ .Values.genesis.network }}?alt=media"
```

Comment:
What do you think about having the genesis/bootnodeEnode be retrieved by celotool from GCS at deploy time? That way if a pod gets restarted and then for some reason these files are inaccessible it wouldn't affect the pod

Reply:
So it should be the case if Values.genesis.useGenesisFileBase64=true. In that case the `config` volume is mounted at `/var/geth` and the genesis is then copied to `/root/.celo`. For the bootnodeEnode, if useGenesisFileBase64=true, should be generated including the `common.celotool-container`, which creates here an unclear dependency. This configuration is currently used for the testnet chart.
Maybe we can generate the bootenode address from celotool and pass it in the same configmap as the genesis file. I think that approach fits better now.
If useGenesisFileBase64!=true, in that case both the genesis and the bootenode is downloaded from gcp. It adds the dependency to download the files from gcp when the container starts, but also I think it helps to decouple the chart from celotool and helps portability. The aks fullnode chart is using this configuration.